### PR TITLE
feat: better config management throughout client

### PR DESCRIPTION
**Pull request by sinasab**
Updating configuration of the client to be much more flexible.

- separating explicit chain vs server config
- making all aspects optional with defaults where needed
- method calls enable overriding via parameter as-needed
- touchup on docs
- bumped some package versions

Tested using beta version @ https://www.npmjs.com/package/@witnessco/client?activeTab=versions

### CODE REVIEW

File: src/defaults.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export const DEFAULT_API_URL = "https://api.witness.co" as const;
+export const defaultServerConfig = {
+	authToken: "",
+	chainId: base.id,
+	endpoint: DEFAULT_API_URL,
+	fetchFn: fetch,
+} as const satisfies ServerConfig;
+
+// Defaults to Base for onchain stuff.
+const DEFAULT_ETH_RPC_URL =
```
Comment (mw2000) [inline src/defaults.ts:15]: We could be exposing our API key here

File: src/WitnessClient.ts
Comment (mw2000) [inline src/WitnessClient.ts:1]: Nice work seperating this out

**Review by mw2000 — CHANGES_REQUESTED** [review]
Just want to make sure that we aren't exposing any internal key and should be gtg

File: src/defaults.ts (hunk 1 above)
Comment (sina-wit) [reply src/defaults.ts:15]: This is expected.

**Review by mw2000 — APPROVED** [review]

**Pull request merged**